lib/blocktree/node.go

```diff
@@ -55,17 +55,24 @@ func (n *node) createTree(tree gotree.Tree) {
 
 // getNode recursively searches for a node with a given hash
 func (n *node) getNode(h common.Hash) *node {
+	if n == nil {
+		return nil
+	}
+
 	if n.hash == h {
 		return n
-	} else if len(n.children) == 0 {
+	}
+
+	if len(n.children) == 0 {
```

Comment:
nit I guess you could remove that if clause, since the for loop below will just not execute below, and return `nil`.

Reply:
done